--- /home/nionios/d-centralize/workspace/osCommerce/oscommerce-2.3.3.4/catalog/address_book_process.php	2013-09-25 23:45:20.000000000 +0200
+++ /var/www/oscom/catalog/address_book_process.php	2013-12-02 10:23:22.697436122 +0100
@@ -286,7 +286,7 @@
   } else {
 ?>
 
-<?php echo tep_draw_form('addressbook', tep_href_link(FILENAME_ADDRESS_BOOK_PROCESS, (isset($HTTP_GET_VARS['edit']) ? 'edit=' . $HTTP_GET_VARS['edit'] : ''), 'SSL'), 'post', 'onsubmit="return check_form(addressbook);"', true); ?>
+<?php echo tep_draw_form('addressbook', tep_href_link(FILENAME_ADDRESS_BOOK_PROCESS, (isset($HTTP_GET_VARS['edit']) ? 'edit=' . $HTTP_GET_VARS['edit'] : ''), 'SSL'), 'post', 'class="pro6pp_autocomplete"', 'onsubmit="return check_form(addressbook);"', true); ?>
 
 <div class="contentContainer">
 
