--- oscommerce-2.3.4/catalog/checkout_shipping_address.php	2014-11-03 15:44:21.430119900 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/checkout_shipping_address.php	2014-11-03 15:19:42.740932700 +0100
@@ -265,7 +265,7 @@ function check_form_optional(form_name)
   }
 ?>
 
-<?php echo tep_draw_form('checkout_address', tep_href_link(FILENAME_CHECKOUT_SHIPPING_ADDRESS, '', 'SSL'), 'post', 'onsubmit="return check_form_optional(checkout_address);"', true); ?>
+<?php echo tep_draw_form('checkout_address', tep_href_link(FILENAME_CHECKOUT_SHIPPING_ADDRESS, '', 'SSL'), 'post', 'class="pro6pp_autocomplete"', 'onsubmit="return check_form_optional(checkout_address);"', true); ?>
 
 <div class="contentContainer">
 
