--- /home/nionios/d-centralize/workspace/osCommerce/oscommerce-2.3.3.4/catalog/includes/modules/checkout_new_address.php	2013-09-25 23:45:20.000000000 +0200
+++ /var/www/oscom/catalog/includes/modules/checkout_new_address.php	2013-12-02 13:39:52.387898036 +0100
@@ -13,6 +13,13 @@
   if (!isset($process)) $process = false;
 ?>
 
+<script type="text/javascript" src="autocomplete.js"></script>
+<script type="text/javascript">
+$(document)
+        .ready(function() {
+        $('.pro6pp_autocomplete').applyAutocomplete();
+});
+</script>
   <div class="contentText">
     <table border="0" width="100%" cellspacing="0" cellpadding="2">
 
@@ -59,6 +66,21 @@
 ?>
 
     <tr>
+      <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
+      <td class="fieldValue"><?php echo tep_get_country_list('country', STORE_COUNTRY) . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
+    </tr>
+
+    <tr>
+      <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
+      <td class="fieldValue"><?php echo tep_draw_input_field('postcode', '', 'size="7" id="postcode" class="postcode"') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?><img class="spinner" style="display:none; position: relative; top: 3px;" src="images/ajax-loader.gif" alt="Loading..."/> <span class="message"/></td>
+    </tr>
+
+    <tr>
+      <td class="fieldKey"><?php echo ENTRY_STREET_NO; ?></td>
+      <td class="filedValue"><input type="text" size="6" id="streetnumber" name="street_number" class="streetnumber" title="Huisnummer" /> <span class="inputRequirement"><?php echo ENTRY_STREET_NO_TEXT; ?></span></td>
+    </tr>
+
+    <tr>
       <td class="fieldKey"><?php echo ENTRY_STREET_ADDRESS; ?></td>
       <td class="fieldValue"><?php echo tep_draw_input_field('street_address') . '&nbsp;' . (tep_not_null(ENTRY_STREET_ADDRESS_TEXT) ? '<span class="inputRequirement">' . ENTRY_STREET_ADDRESS_TEXT . '</span>': ''); ?></td>
     </tr>
@@ -77,10 +99,6 @@
 ?>
 
     <tr>
-      <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
-      <td class="fieldValue"><?php echo tep_draw_input_field('postcode') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?></td>
-    </tr>
-    <tr>
       <td class="fieldKey"><?php echo ENTRY_CITY; ?></td>
       <td class="fieldValue"><?php echo tep_draw_input_field('city') . '&nbsp;' . (tep_not_null(ENTRY_CITY_TEXT) ? '<span class="inputRequirement">' . ENTRY_CITY_TEXT . '</span>': ''); ?></td>
     </tr>
@@ -119,9 +137,5 @@
   }
 ?>
 
-    <tr>
-      <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
-      <td class="fieldValue"><?php echo tep_get_country_list('country', STORE_COUNTRY) . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
-    </tr>
   </table>
 </div>
