--- oscommerce-2.3.4/catalog/includes/languages/english.php	2014-11-03 15:44:21.479124800 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/includes/languages/english.php	2014-11-03 15:18:56.805932700 +0100
@@ -122,6 +122,9 @@ define('ENTRY_EMAIL_ADDRESS_TEXT', '*');
 define('ENTRY_STREET_ADDRESS', 'Street Address:');
 define('ENTRY_STREET_ADDRESS_ERROR', 'Your Street Address must contain a minimum of ' . ENTRY_STREET_ADDRESS_MIN_LENGTH . ' characters.');
 define('ENTRY_STREET_ADDRESS_TEXT', '*');
+define('ENTRY_STREET_NO', 'Street Number:');
+define('ENTRY_STREET_NO_ERROR', 'Your Street number must contain a minimum of 1 characters.');
+define('ENTRY_STREET_NO_TEXT', '*');
 define('ENTRY_SUBURB', 'Suburb:');
 define('ENTRY_SUBURB_TEXT', '');
 define('ENTRY_POST_CODE', 'Post Code:');
