diff -rupN oscommerce-2.3.4/catalog/address_book_process.php pro6pp_oscommerce-2.3.4/catalog/address_book_process.php
--- oscommerce-2.3.4/catalog/address_book_process.php	2014-11-03 15:44:21.417118600 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/address_book_process.php	2014-11-03 15:01:11.323106900 +0100
@@ -286,7 +286,7 @@
   } else {
 ?>
 
-<?php echo tep_draw_form('addressbook', tep_href_link(FILENAME_ADDRESS_BOOK_PROCESS, (isset($HTTP_GET_VARS['edit']) ? 'edit=' . $HTTP_GET_VARS['edit'] : ''), 'SSL'), 'post', 'onsubmit="return check_form(addressbook);"', true); ?>
+<?php echo tep_draw_form('addressbook', tep_href_link(FILENAME_ADDRESS_BOOK_PROCESS, (isset($HTTP_GET_VARS['edit']) ? 'edit=' . $HTTP_GET_VARS['edit'] : ''), 'SSL'), 'post', 'class="pro6pp_autocomplete"', 'onsubmit="return check_form(addressbook);"', true); ?>
 
 <div class="contentContainer">
 
diff -rupN oscommerce-2.3.4/catalog/autocomplete.js pro6pp_oscommerce-2.3.4/catalog/autocomplete.js
--- oscommerce-2.3.4/catalog/autocomplete.js	1970-01-01 01:00:00.000000000 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/autocomplete.js	2014-11-03 15:36:02.086190500 +0100
@@ -0,0 +1,168 @@
+ // Replace this auth_key with your own.
+ var pro6pp_auth_key = "000000000000";
+ 
+if (typeof jQuery == 'undefined') {
+    document
+            .write('<script type="text/javascript" src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.8.0.min.js"></'
+                    + 'script>');
+}
+
+ // Create closure to keep namespace clean and hide implementation.
+ (function($) {
+
+    $.fn.applyAutocomplete = function(applyToClass) {
+
+        var parent_obj = this;
+        NL_SIXPP_REGEX = /[0-9]{4,4}\s?[a-zA-Z]{2,2}/;
+ 
+        parent_obj.postcode = $('input[name=postcode]');
+        parent_obj.streetnumber = $('input[name=street_number]');
+        parent_obj.street = $('input[name=street_address]');
+        parent_obj.city = $('input[name=city]');
+        parent_obj.municipality = $('input[name=suburb]');
+        parent_obj.province = $('input[name=state]');
+        parent_obj.spinner = $('.spinner');
+        parent_obj.message = $('.message');
+        parent_obj.country = $('select[name=country]');
+
+        parent_obj.country.change(function() {
+            var selected = $("select[name=country] :selected").text().trim();
+            if (selected == "Netherlands" || selected == "Nederlands") {
+                parent_obj.postcode.bind("blur", function() {
+                    autocomplete(parent_obj);
+                });
+                parent_obj.streetnumber.bind("keyup", function() {
+                    autocomplete(parent_obj);
+                });
+                fieldsLock(parent_obj, 'apply');
+            } else {
+                parent_obj.postcode.unbind("blur");
+                parent_obj.streetnumber.unbind("keyup");
+                resetFields(parent_obj, 'all');
+                fieldsLock(parent_obj, 'release');
+            }
+        });
+        parent_obj.country.trigger("change");
+    };
+
+    function fieldsLock(obj, action) {
+        if (action == 'release') {
+            obj.postcode.removeAttr('autocomplete');
+            obj.postcode.removeAttr('maxlength');
+            obj.street.removeAttr('readonly');
+            obj.municipality.removeAttr('readonly');
+            obj.city.removeAttr('readonly');
+            obj.province.removeAttr('readonly');
+        } else if (action == 'apply') {
+            obj.postcode.attr('autocomplete', 'off');
+            obj.postcode.attr('maxlength', '7');
+            obj.street.attr('readonly', 'readonly');
+            obj.municipality.attr('readonly', 'readonly');
+            obj.city.attr('readonly', 'readonly');
+            obj.province.attr('readonly', 'readonly');
+        } else
+            return;
+    }
+
+    var pro6pp_cache = {};
+    function pro6pp_cached_get(obj, url, callback) {
+        key = escape(url);
+        if (pro6pp_cache.hasOwnProperty(key)) {
+            callback(obj, pro6pp_cache[key]);
+        } else {
+            $.getJSON(url + "&callback=?", function(data) {
+                pro6pp_cache[key] = data;
+                callback(obj, data);
+            });
+        }
+    }
+
+    function resetFields(obj, all) {
+        all = all || false;
+        if (all) {
+            obj.postcode.val('');
+            obj.streetnumber.val('');
+        }
+        obj.street.val('');
+        obj.city.val('');
+        obj.municipality.val('');
+        obj.province.val('');
+    }
+
+    function resetCompletions(obj) {
+        obj.street.val('');
+        obj.city.val('');
+        obj.municipality.val('');
+        obj.country.val('');
+        obj.province.val('');
+    }
+
+    function autocomplete(obj) {
+        // $(obj).find('.message').html("");
+        var postcode = obj.postcode.val();
+        var streetnumber = obj.streetnumber.val();
+        // Trigger on '5408xb' and on '5408 xb'
+        if (isValidPostcode(obj) && isValidStreetNumber(obj)) {
+            obj.spinner.show();
+            /*
+             * $.getJSON("http://api.pro6pp.nl/v1/autocomplete?auth_key=" +
+             * pro6pp_auth_key + "&nl_sixpp=" + escape(postcode) +
+             * "&streetnumber=" + escape(streetnumber) + "&callback=?",
+             * function(data) { fillin(obj, data); });
+             */
+            var url = "http://api.pro6pp.nl/v1/autocomplete?auth_key="
+                    + pro6pp_auth_key + "&nl_sixpp=" + escape(postcode)
+                    + "&streetnumber=" + escape(streetnumber);
+            pro6pp_cached_get(obj, url, fillin);
+        } else {
+            resetFields(obj);
+        }
+    }
+
+    function isValidPostcode(obj) {
+        var postcode = obj.postcode.val();
+        if (postcode.length) {
+            if (postcode.length == 6 || postcode.length == 7)
+                if (NL_SIXPP_REGEX.test(postcode))
+                    return true;
+                else
+                    obj.message.text('Invalid postcode format.');
+        }
+        return false;
+    }
+
+    function isValidStreetNumber(obj) {
+        var streetnumber = obj.streetnumber.val();
+        var regex = /\d/;
+        if (streetnumber.length >= 1) {
+            if (regex.test(streetnumber))
+                return true;
+            else
+                obj.message.text('Ivalid streetnumber format.');
+        }
+        return false;
+    }
+
+    function fillin(obj, json) {
+        obj.spinner.hide();
+        if (json.status == 'ok') {
+            if (json.results.length >= 1) {
+                obj.street.val(json.results[0].street);
+                obj.city.val(json.results[0].city);
+                obj.municipality.val(json.results[0].municipality);
+                obj.province.val(json.results[0].province);
+                obj.message.hide();
+            }
+        } else {
+            var translated_message = json.error.message;
+            if (json.error.message == 'nl_sixpp not found') {
+                translated_message = 'Onbekende postcode';
+            } else if (json.error.message == 'invalid postcode format') {
+                translated_message = 'Ongeldig postcode formaat';
+            }
+            resetFields(obj);
+            obj.message.text(translated_message);
+            obj.message.show();
+        }
+    }
+})($);
diff -rupN oscommerce-2.3.4/catalog/checkout_shipping_address.php pro6pp_oscommerce-2.3.4/catalog/checkout_shipping_address.php
--- oscommerce-2.3.4/catalog/checkout_shipping_address.php	2014-11-03 15:44:21.430119900 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/checkout_shipping_address.php	2014-11-03 15:19:42.740932700 +0100
@@ -265,7 +265,7 @@ function check_form_optional(form_name)
   }
 ?>
 
-<?php echo tep_draw_form('checkout_address', tep_href_link(FILENAME_CHECKOUT_SHIPPING_ADDRESS, '', 'SSL'), 'post', 'onsubmit="return check_form_optional(checkout_address);"', true); ?>
+<?php echo tep_draw_form('checkout_address', tep_href_link(FILENAME_CHECKOUT_SHIPPING_ADDRESS, '', 'SSL'), 'post', 'class="pro6pp_autocomplete"', 'onsubmit="return check_form_optional(checkout_address);"', true); ?>
 
 <div class="contentContainer">
 
diff -rupN oscommerce-2.3.4/catalog/create_account.php pro6pp_oscommerce-2.3.4/catalog/create_account.php
--- oscommerce-2.3.4/catalog/create_account.php	2014-11-03 15:44:21.467123600 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/create_account.php	2014-11-03 15:20:05.843932700 +0100
@@ -258,6 +258,14 @@
   require('includes/form_check.js.php');
 ?>
 
+<script type="text/javascript" src="autocomplete.js"></script>
+<script type="text/javascript">
+$(document)
+        .ready(function() {
+        $('.pro6pp_autocomplete').applyAutocomplete();
+});
+</script>
+
 <h1><?php echo HEADING_TITLE; ?></h1>
 
 <?php
@@ -268,7 +276,7 @@
 
 <p><?php echo sprintf(TEXT_ORIGIN_LOGIN, tep_href_link(FILENAME_LOGIN, tep_get_all_get_params(), 'SSL')); ?></p>
 
-<?php echo tep_draw_form('create_account', tep_href_link(FILENAME_CREATE_ACCOUNT, '', 'SSL'), 'post', 'onsubmit="return check_form(create_account);"', true) . tep_draw_hidden_field('action', 'process'); ?>
+<?php echo tep_draw_form('create_account', tep_href_link(FILENAME_CREATE_ACCOUNT, (isset($HTTP_GET_VARS['guest'])? 'guest=guest':''), 'SSL'), 'post', 'class="pro6pp_autocomplete"', 'onsubmit="return check_form(create_account);"', true) . tep_draw_hidden_field('action', 'process'); ?>
 
 <div class="contentContainer">
   <div>
@@ -296,7 +304,7 @@
         <td class="fieldKey"><?php echo ENTRY_FIRST_NAME; ?></td>
         <td class="fieldValue"><?php echo tep_draw_input_field('firstname') . '&nbsp;' . (tep_not_null(ENTRY_FIRST_NAME_TEXT) ? '<span class="inputRequirement">' . ENTRY_FIRST_NAME_TEXT . '</span>': ''); ?></td>
       </tr>
-      <tr> 
+      <tr>
         <td class="fieldKey"><?php echo ENTRY_LAST_NAME; ?></td>
         <td class="fieldValue"><?php echo tep_draw_input_field('lastname') . '&nbsp;' . (tep_not_null(ENTRY_LAST_NAME_TEXT) ? '<span class="inputRequirement">' . ENTRY_LAST_NAME_TEXT . '</span>': ''); ?></td>
       </tr>
@@ -344,6 +352,20 @@
 
   <div class="contentText">
     <table border="0" cellspacing="2" cellpadding="2" width="100%">
+
+      <tr>
+        <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
+        <td class="fieldValue"><?php echo tep_get_country_list('country') . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
+      </tr>
+
+      <tr>
+        <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
+        <td class="fieldValue"><?php echo tep_draw_input_field('postcode', '', 'size="7" id="postcode" class="postcode"') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?><img class="spinner" style="display:none; position: relative; top: 3px;" src="images/ajax-loader.gif" alt="Loading..."/> <span class="message"/></td>
+      </tr>
+      <tr>
+      <td class="fieldKey"><?php echo ENTRY_STREET_NO; ?></td>
+      <td class="filedValue"><input type="text" size="6" id="streetnumber" name="street_number" class="streetnumber" title="Huisnummer" /> <span class="inputRequirement"><?php echo ENTRY_STREET_NO_TEXT; ?></span></td>
+     </tr>
       <tr>
         <td class="fieldKey"><?php echo ENTRY_STREET_ADDRESS; ?></td>
         <td class="fieldValue"><?php echo tep_draw_input_field('street_address') . '&nbsp;' . (tep_not_null(ENTRY_STREET_ADDRESS_TEXT) ? '<span class="inputRequirement">' . ENTRY_STREET_ADDRESS_TEXT . '</span>': ''); ?></td>
@@ -361,11 +383,6 @@
 <?php
   }
 ?>
-
-      <tr>
-        <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
-        <td class="fieldValue"><?php echo tep_draw_input_field('postcode') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?></td>
-      </tr>
       <tr>
         <td class="fieldKey"><?php echo ENTRY_CITY; ?></td>
         <td class="fieldValue"><?php echo tep_draw_input_field('city') . '&nbsp;' . (tep_not_null(ENTRY_CITY_TEXT) ? '<span class="inputRequirement">' . ENTRY_CITY_TEXT . '</span>': ''); ?></td>
@@ -403,10 +420,6 @@
   }
 ?>
 
-      <tr>
-        <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
-        <td class="fieldValue"><?php echo tep_get_country_list('country') . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
-      </tr>
     </table>
   </div>
 
diff -rupN oscommerce-2.3.4/catalog/includes/languages/english.php pro6pp_oscommerce-2.3.4/catalog/includes/languages/english.php
--- oscommerce-2.3.4/catalog/includes/languages/english.php	2014-11-03 15:44:21.479124800 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/includes/languages/english.php	2014-11-03 15:18:56.805932700 +0100
@@ -122,6 +122,9 @@ define('ENTRY_EMAIL_ADDRESS_TEXT', '*');
 define('ENTRY_STREET_ADDRESS', 'Street Address:');
 define('ENTRY_STREET_ADDRESS_ERROR', 'Your Street Address must contain a minimum of ' . ENTRY_STREET_ADDRESS_MIN_LENGTH . ' characters.');
 define('ENTRY_STREET_ADDRESS_TEXT', '*');
+define('ENTRY_STREET_NO', 'Street Number:');
+define('ENTRY_STREET_NO_ERROR', 'Your Street number must contain a minimum of 1 characters.');
+define('ENTRY_STREET_NO_TEXT', '*');
 define('ENTRY_SUBURB', 'Suburb:');
 define('ENTRY_SUBURB_TEXT', '');
 define('ENTRY_POST_CODE', 'Post Code:');
diff -rupN oscommerce-2.3.4/catalog/includes/modules/address_book_details.php pro6pp_oscommerce-2.3.4/catalog/includes/modules/address_book_details.php
--- oscommerce-2.3.4/catalog/includes/modules/address_book_details.php	2014-11-03 15:44:21.490125900 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/includes/modules/address_book_details.php	2014-11-03 15:17:55.319932700 +0100
@@ -13,6 +13,13 @@
   if (!isset($process)) $process = false;
 ?>
 
+<script type="text/javascript" src="autocomplete.js"></script>
+<script type="text/javascript">
+$(document)
+        .ready(function() {
+        $('.pro6pp_autocomplete').applyAutocomplete();
+});
+</script>
   <div>
     <span class="inputRequirement" style="float: right;"><?php echo FORM_REQUIRED_INFORMATION; ?></span>
     <h2><?php echo NEW_ADDRESS_TITLE; ?></h2>
@@ -65,6 +72,21 @@
 ?>
 
       <tr>
+        <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
+        <td class="fieldValue"><?php echo tep_get_country_list('country', (isset($entry['entry_country_id']) ? $entry['entry_country_id'] : STORE_COUNTRY)) . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
+      </tr>
+
+      <tr>
+        <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
+        <td class="fieldValue"><?php echo tep_draw_input_field('postcode', (isset($entry['entry_postcode']) ? $entry['entry_postcode'] : ''), 'size="7" id="postcode" class="postcode"') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?><img class="spinner" style="display:none; position: relative; top: 3px;" src="images/ajax-loader.gif" alt="Loading..."/> <span class="message"/></td>
+      </tr>
+
+      <tr>
+        <td class="fieldKey"><?php echo ENTRY_STREET_NO; ?></td>
+        <td class="filedValue"><input type="text" size="6" id="streetnumber" name="street_number" class="streetnumber" title="Huisnummer" /> <span class="inputRequirement"><?php echo ENTRY_STREET_NO_TEXT; ?></span></td>
+      </tr>
+
+      <tr>
         <td class="fieldKey"><?php echo ENTRY_STREET_ADDRESS; ?></td>
         <td class="fieldValue"><?php echo tep_draw_input_field('street_address', (isset($entry['entry_street_address']) ? $entry['entry_street_address'] : '')) . '&nbsp;' . (tep_not_null(ENTRY_STREET_ADDRESS_TEXT) ? '<span class="inputRequirement">' . ENTRY_STREET_ADDRESS_TEXT . '</span>': ''); ?></td>
       </tr>
@@ -83,10 +105,6 @@
 ?>
 
       <tr>
-        <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
-        <td class="fieldValue"><?php echo tep_draw_input_field('postcode', (isset($entry['entry_postcode']) ? $entry['entry_postcode'] : '')) . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?></td>
-      </tr>
-      <tr>
         <td class="fieldKey"><?php echo ENTRY_CITY; ?></td>
         <td class="fieldValue"><?php echo tep_draw_input_field('city', (isset($entry['entry_city']) ? $entry['entry_city'] : '')) . '&nbsp;' . (tep_not_null(ENTRY_CITY_TEXT) ? '<span class="inputRequirement">' . ENTRY_CITY_TEXT . '</span>': ''); ?></td>
       </tr>
@@ -123,10 +141,6 @@
   }
 ?>
 
-      <tr>
-        <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
-        <td class="fieldValue"><?php echo tep_get_country_list('country', (isset($entry['entry_country_id']) ? $entry['entry_country_id'] : STORE_COUNTRY)) . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
-      </tr>
 
 <?php
   if ((isset($HTTP_GET_VARS['edit']) && ($customer_default_address_id != $HTTP_GET_VARS['edit'])) || (isset($HTTP_GET_VARS['edit']) == false) ) {
diff -rupN oscommerce-2.3.4/catalog/includes/modules/checkout_new_address.php pro6pp_oscommerce-2.3.4/catalog/includes/modules/checkout_new_address.php
--- oscommerce-2.3.4/catalog/includes/modules/checkout_new_address.php	2014-11-03 15:44:21.500126900 +0100
+++ pro6pp_oscommerce-2.3.4/catalog/includes/modules/checkout_new_address.php	2014-11-03 15:18:21.863932700 +0100
@@ -13,6 +13,13 @@
   if (!isset($process)) $process = false;
 ?>
 
+<script type="text/javascript" src="autocomplete.js"></script>
+<script type="text/javascript">
+$(document)
+        .ready(function() {
+        $('.pro6pp_autocomplete').applyAutocomplete();
+});
+</script>
   <div class="contentText">
     <table border="0" width="100%" cellspacing="0" cellpadding="2">
 
@@ -59,6 +66,21 @@
 ?>
 
     <tr>
+      <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
+      <td class="fieldValue"><?php echo tep_get_country_list('country', STORE_COUNTRY) . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
+    </tr>
+
+    <tr>
+      <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
+      <td class="fieldValue"><?php echo tep_draw_input_field('postcode', '', 'size="7" id="postcode" class="postcode"') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?><img class="spinner" style="display:none; position: relative; top: 3px;" src="images/ajax-loader.gif" alt="Loading..."/> <span class="message"/></td>
+    </tr>
+
+    <tr>
+      <td class="fieldKey"><?php echo ENTRY_STREET_NO; ?></td>
+      <td class="filedValue"><input type="text" size="6" id="streetnumber" name="street_number" class="streetnumber" title="Huisnummer" /> <span class="inputRequirement"><?php echo ENTRY_STREET_NO_TEXT; ?></span></td>
+    </tr>
+
+    <tr>
       <td class="fieldKey"><?php echo ENTRY_STREET_ADDRESS; ?></td>
       <td class="fieldValue"><?php echo tep_draw_input_field('street_address') . '&nbsp;' . (tep_not_null(ENTRY_STREET_ADDRESS_TEXT) ? '<span class="inputRequirement">' . ENTRY_STREET_ADDRESS_TEXT . '</span>': ''); ?></td>
     </tr>
@@ -77,10 +99,6 @@
 ?>
 
     <tr>
-      <td class="fieldKey"><?php echo ENTRY_POST_CODE; ?></td>
-      <td class="fieldValue"><?php echo tep_draw_input_field('postcode') . '&nbsp;' . (tep_not_null(ENTRY_POST_CODE_TEXT) ? '<span class="inputRequirement">' . ENTRY_POST_CODE_TEXT . '</span>': ''); ?></td>
-    </tr>
-    <tr>
       <td class="fieldKey"><?php echo ENTRY_CITY; ?></td>
       <td class="fieldValue"><?php echo tep_draw_input_field('city') . '&nbsp;' . (tep_not_null(ENTRY_CITY_TEXT) ? '<span class="inputRequirement">' . ENTRY_CITY_TEXT . '</span>': ''); ?></td>
     </tr>
@@ -119,9 +137,5 @@
   }
 ?>
 
-    <tr>
-      <td class="fieldKey"><?php echo ENTRY_COUNTRY; ?></td>
-      <td class="fieldValue"><?php echo tep_get_country_list('country', STORE_COUNTRY) . '&nbsp;' . (tep_not_null(ENTRY_COUNTRY_TEXT) ? '<span class="inputRequirement">' . ENTRY_COUNTRY_TEXT . '</span>': ''); ?></td>
-    </tr>
   </table>
 </div>
